convert2rhel/actions/system_checks/convert2rhel_latest.py

```diff
@@ -0,0 +1,196 @@
+
+        if not system_info.has_internet_access:
+            logger.warning("Skipping the check because no internet connection has been detected.")
+            return
+
+        repo_dir = tempfile.mkdtemp(prefix="convert2rhel_repo.", dir=utils.TMP_DIR)
+        repo_path = os.path.join(repo_dir, "convert2rhel.repo")
+        utils.store_content_to_file(filename=repo_path, content=CONVERT2RHEL_REPO_CONTENT)
+
+        cmd = [
+            "repoquery",
+            "--disablerepo=*",
+            "--enablerepo=convert2rhel",
+            "--releasever=%s" % system_info.version.major,
+            "--setopt=reposdir=%s" % repo_dir,
+            "--qf",
+            "C2R %{NAME}-%{EPOCH}:%{VERSION}-%{RELEASE}.%{ARCH}",
+            "convert2rhel",
+        ]
+
+        # Note: This is safe because we're creating in utils.TMP_DIR which is hardcoded to
+        # /var/lib/convert2rhel which does not have any world-writable directory components.
+        utils.mkdir_p(repo_dir)
+
+        try:
+            raw_output_convert2rhel_versions, return_code = utils.run_subprocess(cmd, print_output=False)
+        finally:
+            shutil.rmtree(repo_dir)
+
+        if return_code != 0:
+            logger.warning(
+                "Couldn't check if the current installed Convert2RHEL is the latest version.\n"
+                "repoquery failed with the following output:\n%s" % (raw_output_convert2rhel_versions)
+            )
+            return
+
+        # convert the raw output of convert2rhel version strings into a list
+        raw_output_convert2rhel_versions = raw_output_convert2rhel_versions.splitlines()
+
+        temp_raw_output = []
+
+        # We are expecting an repoquery output to be similar to this:
+        # C2R convert2rhel-0:0.17-1.el7.noarch
+        # We need the `C2R` identifier to be present on the line so we can know for
+        # sure that the line we are working with is the a line that contains
+        # relevant repoquery information to our check, otherwise, we just log the
+        # information as debug and do nothing with it.
+        for raw_version in raw_output_convert2rhel_versions:
+            if "C2R" in raw_version:
+                temp_raw_output.append(raw_version.lstrip("C2R "))
+            else:
+                # Mainly for debugging purposes to see what is happening if we got
+                # anything else that does not have the C2R identifier at the start
+                # of the line.
+                logger.debug("Got a line without the C2R identifier: %s" % raw_version)
+        raw_output_convert2rhel_versions = temp_raw_output
+
+        latest_available_version = ("0", "0.00", "0")
+        convert2rhel_versions = []
+
+        # add each tuple of fields obtained from parse_pkg_string() to convert2rhel_versions
+        for raw_pkg in raw_output_convert2rhel_versions:
+            try:
+                parsed_pkg = parse_pkg_string(raw_pkg)
+
+            except ValueError as exc:
```

Comment:
```suggestion
            except ValueError as exc:
```

Reply:
blank space between line and exception

Reply:
Note: If the `try:` block s long enough, I think having a blank line above the `except:` is good for readability.  Since this is a single line `try:` block, that doesn't apply here.